### PR TITLE
Feature/Creates Searchbar Component

### LINKED ISSUE #311: App: Create SearchBar component
We need to create a search bar that allows the user to search for postings or other users. When they type in a search phrase, we can make a api calls with the search, and we'll get back a list of postings or users matching the search text.

### PR DESCRIPTION
**Pull request by trentstauff**
Closes #311 

This feature implements a global uMigrate search for users to look up any type of post, event, user, poll etc by entering some text.

# FEATURES
- Adds ability to enter search view from each individual "page" (community, market, housing), which currently sits as the header to the FeedContainer flatlist.

- When first entering the search view, the page will display the 5 most recent posts from each type.
- When the user types and finishes something, the page will automatically update to show the results, dynamically showcasing only sections that have results.
- Supports pagination of results so the API is not overwhelmed, so when the user presses "More" the next page (if available) will be loaded
- Has a "Collapse" button that allows that section's results (if more than one page is loaded) to be collapsed back to the top initial results
- Lets users know when there is no more results for a particular topic

https://user-images.githubusercontent.com/53923200/118418149-3588de80-b685-11eb-869e-4110c3e0e224.mov

### CODE REVIEW

File: app/src/src/components/containers/SearchResults.jsx [hunk 1]
```diff
@@ -0,0 +1,268 @@
+import React, { useState, useEffect, useCallback } from 'react';
```
Comment (trentstauff) [inline app/src/src/components/containers/SearchResults.jsx:1]: Ive noticed i have quite inconsistent variable names, ill look into this

File: app/src/src/components/containers/SearchResults.jsx [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+        // Append error messages to state
+        setErrors([...errors, JSON.stringify(error)]);
+      }
+    }
+    setData(responseDataList);
+  };
+
+  // when triggered, attempts to get more data.
+  get_more = async (endpoint_name, current_page) => {
+    // get copy of current data
```
Comment (trentstauff) [inline app/src/src/components/containers/SearchResults.jsx:148]: shallow copy, should comment

File: app/src/src/components/containers/SearchResults.jsx (hunk 1 above)
Comment (deandrebaker) [reply app/src/src/components/containers/SearchResults.jsx:1]: What do you mean? These are built-in functions

Comment (trentstauff) [reply app/src/src/components/containers/SearchResults.jsx:1]: I just mean in general im all over the place, not specifically here

**Review by deandrebaker — APPROVED** [review]

**Review by deandrebaker — APPROVED** [review]

**Pull request merged**